vsphere/internal/helper/network/network_helper.go

```diff
@@ -32,6 +33,59 @@ func FromPath(client *govmomi.Client, name string, dc *object.Datacenter) (objec
 	return finder.Network(ctx, name)
 }
 
+func FromNameAndDVSUuid(client *govmomi.Client, name string, dc *object.Datacenter, dvsUuid string) (object.NetworkReference, error) {
+
+	finder := find.NewFinder(client.Client, false)
+	if dc != nil {
```

Comment:
What if (dc == nil)? You will continue with a finder without a dc set.

Reply:
DC is optional here. So we can continue without DC too.